src/client.cpp

```diff
@@ -194,22 +194,25 @@ bool hasIntMember(const rapidjson::Value& value, const char* name)
     return value.HasMember(name)  &&  value[name].IsInt();
 }
 
+#define CREATE_SESSION() \
+    CurlSessionPtr sessionPtr = createSession();\
+    if (!sessionPtr)\
+        return makeError();\
+    CurlSession& session = *sessionPtr;
```

Comment:
Please see [here](http://stackoverflow.com/questions/154136/why-use-apparently-meaningless-do-while-and-if-else-statements-in-c-c-macros).